### PR TITLE
[MockAPI] fix the issue when session gets updated with an invalid value of `name` 

### PR DESCRIPTION
**Pull request by PeterSenpai**
[issue#317](https://github.com/uoft-tapp/tapp/issues/317)

### CODE REVIEW

**Review by colezemel — APPROVED** [review]
Looks good!

File: frontend/src/api/mockAPI/sessions.js [hunk 1]
```diff
@@ -26,6 +26,36 @@ export class Session extends MockAPIController {
             throw new Error(message);
         }
     }
+    validateUpdate(session) {
+        if ("name" in session) {
+            if (session.name === undefined || session.name.length === 0) {
```
Comment (CarolineYao) [inline frontend/src/api/mockAPI/sessions.js:31]: If `session.name` is `undefined`, doesn't `session.name.length` give a TypeError?

**Review by CarolineYao — APPROVED** [review]

File: frontend/src/api/mockAPI/sessions.js [hunk 2]
```diff
@@ -26,6 +26,36 @@ export class Session extends MockAPIController {
             throw new Error(message);
         }
     }
+    validateUpdate(session) {
```
Comment (PeterSenpai) [inline frontend/src/api/mockAPI/sessions.js]: don't create a new method here

File: frontend/src/api/mockAPI/utils.js [hunk 3]
```diff
@@ -134,7 +134,10 @@ export function getAttributesCheckMessage(
 ) {
     for (const [prop, requirements] of Object.entries(props)) {
         // Required attributes cannot be null or the empty string
-        if (requirements.required && (obj[prop] == null || obj[prop] === "")) {
+        if (
```
Comment (PeterSenpai) [inline frontend/src/api/mockAPI/utils.js]: undo the change and make a comment

File: frontend/src/api/mockAPI/utils.js [hunk 4]
```diff
@@ -245,6 +246,18 @@ export class MockAPIController {
...
+    /**
+     * Validates a property of an item instance. Throws an error
+     * if this property are invalid/incompatable.
+     *
+     * @param {object} obj
+     * @param {string} prop
+     * @memberof MockAPIController
+     */
+    validateProp(obj, prop) {
+        if (obj[prop]) return true;
```
Comment (siefkenj) [inline frontend/src/api/mockAPI/utils.js]: Never, ever, ever use an if statement without `{...}` around it's following block. Also, this code is wrong. You probably meant `prop in obj` or `obj[prop] != null`.

Also, why the if statement? I think the if statement should not be there.

Comment (PeterSenpai) [reply frontend/src/api/mockAPI/utils.js]: I see, if I don't use obj or prop anywhere inside this function, prettier would complain(fail the formatting tests). but both obj and prop are checked before even call this function. what do you think I should do here

Comment (siefkenj) [reply frontend/src/api/mockAPI/utils.js]: You can remove `obj` and `prop` from the function signature. Or you can add a `// eslint-ignore-next-line` (or whatever it is..) comment.

File: frontend/src/api/mockAPI/utils.js [hunk 5]
```diff
@@ -245,6 +246,19 @@ export class MockAPIController {
...
+    /**
+     * Validates a property of an item instance. Throws an error
+     * if this property are invalid/incompatable.
+     *
+     * @param {object} obj
+     * @param {string} prop
+     * @memberof MockAPIController
+     */
+    // eslint-disable-next-line no-unused-vars
+    validateProp(obj, prop) {
```
Comment (siefkenj) [inline frontend/src/api/mockAPI/utils.js]: Why are you passing in an object? I don't get it...

File: frontend/src/api/mockAPI/sessions.js [hunk 6]
```diff
@@ -26,6 +26,28 @@ export class Session extends MockAPIController {
...
         }
     }
+
+    validateProp(obj, prop) {
+        if (prop === "name") {
+            if (obj[prop] === undefined || obj[prop].length === 0) {
+                throw new Error(
+                    `Property ${prop} cannot be empty or undefined.`
+                );
+            } else {
```
Comment (siefkenj) [inline frontend/src/api/mockAPI/sessions.js]: I don't understand what you're doing. I suspect it's wrong (i.e., it doesn't belong in this function). You need comments to explain.

Comment (PeterSenpai) [reply frontend/src/api/mockAPI/sessions.js]: say we want to update a session. `obj` here is the request body `{ id: 1, name: xxx}`. `prop` is the key/field. if the field is `name` , we need to make sure `{ ...,name: ""}` and `{..., name: undefined}` throws an error. And if `name` is not empty, we still need to check if it's updatable, ie, if it is unique to the rest of the sessions (all the session that id is not 1)

File: frontend/src/api/mockAPI/utils.js (hunk 5 above)
Comment (PeterSenpai) [reply frontend/src/api/mockAPI/utils.js]: say we want to update a session. `obj` here is the request body `{ id: xxx, name: xxx}`. `prop` is the key/field. I need to get the value of the field.

File: frontend/src/api/mockAPI/sessions.js (hunk 6 above)
Comment (siefkenj) [reply frontend/src/api/mockAPI/sessions.js]: This function should only check the `prop` that is passed in and nothing else.

File: frontend/src/api/mockAPI/utils.js (hunk 5 above)
Comment (siefkenj) [reply frontend/src/api/mockAPI/utils.js]: You should be passing in the prop and the value, not the prop and the object.

Comment (PeterSenpai) [reply frontend/src/api/mockAPI/utils.js]: right... That sounds way better

File: frontend/src/api/mockAPI/sessions.js (hunk 6 above)
Comment (PeterSenpai) [reply frontend/src/api/mockAPI/sessions.js]: Hmm... You mean it shouldn't throw errors but return false or a message?
